include/aspect/compat.h

```diff
@@ -24,40 +24,43 @@
 #include <aspect/global.h>
 
 
+/*
+ * MPI::min() functions
+ */
+#if DEAL_II_VERSION_GTE(8,3,0)
+// use deal.II's MPI::min() functions
+#else
 namespace dealii
 {
   namespace Utilities
   {
     namespace MPI
     {
-      namespace
+      template <typename T>
+      inline
+      T min (const T &t,
+             const MPI_Comm &mpi_communicator)
       {
-
-        inline MPI_Datatype mpi_type_id (const unsigned int *)
-        {
-          return MPI_UNSIGNED;
-        }
-
-
-        inline MPI_Datatype mpi_type_id (const unsigned long int *)
-        {
-          return MPI_UNSIGNED_LONG;
-        }
+        return -max(t, mpi_communicator);
```

Comment:
Uh, that's not the minimum. You probably mean `-max(-t)`.

To make this always work, could you `#include <deal.II/base/mpi.h>` at the top? Or do you presume that the includer of this file also includes that automatically?